tools/autograd/gen_variable_type.py

```diff
@@ -600,6 +666,28 @@ def wrap_view_single(output_var, base_var):
         else:
             return 'as_variable({})'.format(call), []
 
+    def enforce_same_tensorimpl_and_storage(env, call):
+        pre_call_block = ''
+        post_call_block = ''
+        if declaration['name'] not in DONT_ENFORCE_SAME_TENSOR_IMPL_OR_STORAGE:
+            for arg in env.get('unpacked_args', []):
+                dynamic_type = env['unpacked_args_dynamic_type'][arg]
```

Comment:
is there a 'simple_type' you can use here?  I forget the details, mainly trying to avoid having to check IntegerTensor, etc. rather than Tensor.